codegen/java-gen/src/main/scala/kalix/codegen/java/ComponentsSourceGenerator.scala

```diff
@@ -173,6 +176,16 @@ object ComponentsSourceGenerator {
        |    return context.getComponentGrpcClient(serviceClass);
        |  }
        |
+       |  private <Req, Res> SingleResponseRequestBuilder<Req, Res> addHeaders(SingleResponseRequestBuilder<Req, Res> requestBuilder, Metadata metadata){
+       |    var updatedBuilder = requestBuilder;
+       |    for (Metadata.MetadataEntry entry: metadata){
+       |      if (entry.isText()) {
```

Comment:
Actually, maybe for grpc based sdk, I should support binary headers as well?

Reply:
I think we can skip those for now.